### DISCUSSION TITLE
Generic Titles v frustrating

### DISCUSSION THREAD

**Discussion by g33khubber** [aclap-dev/video-downloadhelper#1014] [Help and Q&A]
### Your question

1. Is it possible to always fetch the video title rather than using a generic name like "MASTER M3U8"? This happens to some of the videos in Teachable. Screenshot for reference: https://i.imgur.com/Qn4cbC8.png
2. For Teachable or other platforms, how do I batch download without clicking into each of the videos and starting the video?

### Addon & CoApp version and details

version: 9.0.2.12
target: google
channel: stable
lang: en-GB
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"e6878406c7034afc"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "download_directory": "D:\\Downloads",
    "concurrent_downloads_max": 20,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 9,
    "successfull_dl": 7
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

For your naming issue, you should use the `Download As` function of VDH instead of the `Download` function.  That way you will have the opportunity to change the file name of what you download.

If you want a way to get VDH to automatically choose something better, it's broken at the moment.  Search this forum with `smartnaming` as your search key.  You'll find other discussions on this relevant topic.

There is no batch download feature in VDH.  You have to manually launch each download.  At least this new user interface in VDH 9 can be operated strictly from the keyboard.  For me, that is the single best improvement VDH 9 has brought.  A combination of Tab, arrow keys, page keys, Spacebar, & Enter do the trick.  Experiment.  You'll get the hang of it.

**Reply by g33khubber** [reply]
ah I see. will you know if there's any ETA on the fix for smart naming? 😔

**Reply by GrampaWildWilly** [reply]
If you read a couple of those other discussions, you'll know we're all in the dark on that.  The only dev is @paulrouget Paul & he's not one to give that sort of game away.

**Reply by paulrouget** [reply]
Working on it! Fix should come in v9.1.

**Comment by paulrouget** [comment]
> * Is it possible to always fetch the video title rather than using a generic name like "MASTER M3U8"? This happens to some of the videos in Teachable.

It should do that normally. Please share an URL where it doesn't do that (please give me an exact URL, not just "any video on teachable").

> * For Teachable or other platforms, how do I batch download without clicking into each of the videos and starting the video?

Most of the time, clicking is not necessary. But more and more platforms only make the video available only after the play button is clicked. Nothing we can do about that.

**Reply by g33khubber** [reply]
link: https://vod-akm.play.hotmart.com/video/5Z36M049La/hls/1080/playlist-1080-t-1715055238001.m3u8?hdntl=exp=1721360110~acl=/*~data=hdntl~hmac=79984e6633ed38a27fdc344eb1f6a8a0299d9e23698ddb461181cb0b6678f47f&app=aa2d356b-e2f0-45e8-9725-e0efc7b5d29c

screenshot: https://i.imgur.com/Rsy1PBm.png

**Reply by paulrouget** [reply]
Thank you!

**Reply by GrampaWildWilly** [reply]
Now I'm a bit puzzled.  When I click that vod-akm.play.hotmart.com URL you posted, it tells me I don't have permission to access the site.  That's not surprising since I'm not a paid member of the site.  But when you click that, do you get a web page with a video on it?  Or do you get some other result?

**Reply by g33khubber** [reply]
> Now I'm a bit puzzled. When I click that vod-akm.play.hotmart.com URL you posted, it tells me I don't have permission to access the site. That's not surprising since I'm not a paid member of the site. But when you click that, do you get a web page with a video on it? Or do you get some other result?

I get Access denied for the direct video link too, but the embedded video on the Teachable site works as intended. 

Would also like to report that downloading of Loom videos do not work too - https://i.imgur.com/n6pglhU.png
https://www.loom.com/share/6dbd7409a99b474e99653d68d6fc4a45

Access Denied
You don't have permission to access "http://vod-akm.play.hotmart.com/video/5Z36M049La/hls/1080/playlist-1080-t-1715055238001.m3u8?" on this server.
Reference #18.9d84dead.1721291865.7ad53e59

https://errors.edgesuite.net/18.9d84dead.1721291865.7ad53e59

**Reply by GrampaWildWilly** [reply]
I got errors on the Loom video, too.  VDH proposed this for me:

![#001](https://github.com/user-attachments/assets/38b8b99f-1266-4449-94fd-348e9c28a3b5)
 
When I tried to download the first RawHLS variant, I got the generic "please report this" error from VDH.  When I tried to download the HTTPMedia variant, it succeeded but all I got was a 2-second clip of the video.

Trying to go to Plan B . . .

![#002](https://github.com/user-attachments/assets/a2c30960-5901-4820-a699-0ae8bf77afb3)

I ran the HLS master manifest through ffprobe & it just gave me 403 Forbidden errors.  I then tried running one of the HLS stream manifests through ffprobe & got the same 403 Forbidden errors.  Strangely, you can get the captions for this.  You can see 2 entries in the Network Monitor for items of type vtt.  If you double click on either one of those, Firefox will open that in its own tab.  It turns out those are not WebVTT captions but srt captions.  You could copy/paste those browser tabs into a plain text file on your system & name it something.srt.  Then VLC would pick it up & display the subtitles . . .  That is, if you could download the video & call it something.mp4.  But that part of the exercise is not cooperating.

I think this is stealth DRM.  The web site is not implementing overt DRM.  But they are putting their manifests behind a user ID & password.  And what are those user ID & password?  Of course, that's not published information.  So it amounts to DRM without actually being DRM.

Unless Paul knows some trick to get around the 403 error, I'd say the Loom stuff is doomed.

**Reply by paulrouget** [reply]
Loom is a known problem (which I know how to fix, but just need to find the time - in my todolist).

**Comment by Mattmjs** [comment]
Another tool **yt-dlp** managed to dodge those errors. This is the formats it found

[loom] Extracting URL: https://www.loom.com/share/6dbd7409a99b474e99653d68d6fc4a45
[loom] 6dbd7409a99b474e99653d68d6fc4a45: Downloading GraphQL metadata JSON
[loom] 6dbd7409a99b474e99653d68d6fc4a45: Downloading GraphQL JSON
[loom] 6dbd7409a99b474e99653d68d6fc4a45: Downloading raw-url JSON
[loom] 6dbd7409a99b474e99653d68d6fc4a45: Downloading m3u8 information
[loom] 6dbd7409a99b474e99653d68d6fc4a45: Downloading transcoded-url JSON
[info] Available formats for 6dbd7409a99b474e99653d68d6fc4a45:
ID              EXT RESOLUTION │   FILESIZE   TBR PROTO │ VCODEC  ACODEC
─────────────────────────────────────────────────────────────────────────
http-transcoded mp4 1920x1080  │                  https │ unknown unknown
hls-raw-1500    mp4 1280x720   │ ~248.73MiB 1500k m3u8  │ unknown unknown
hls-raw-3200    mp4 1920x1080  │ ~530.62MiB 3200k m3u8  │ unknown unknown

I tried the hls-raw-1500 version, the speed was slow, the http-transcoded mp4 was much faster.

**Reply by paulrouget** [reply]
Yes - loom used a non-standard http parameter system. I know how to fix this… just need to find the time 😅
